documentation/docs/examples/timeline.mdx

```diff
@@ -0,0 +1,82 @@
+---
+id: timelinetool
+title: TimeLine Tool
+describe: Applying styles to nodes based on date with Play & Stop
+---
+
+import { DemoWrapper } from '@site/docs/components/DemoWrapper';
+
+Reveal allows for applying styles to nodes based on date property and can be viewed as a playable entity
+where node styles are applied to different nodes which can be used to showcase progress of the model construction.
+
+
+:::note
+The test data used in the documentation site is for testing and to show the capabilities of the tool.
+The examples shown is therefore might not be a ideal model construction progression.
+:::
+
+<DemoWrapper name="Cognite3DViewerDemo" />
+
+## TimeLine
+
+TimeLine tool provides ablility to apply styles to nodes using Date as a property. It requires Time Keyframes to be created.
+Each timeframe consists of nodes which are the elements to be rendered, appearance/styles to be assigned to the nodes and
+lastly the date for the nodes as reference to be rendered.
+
+### Create TimeFrame
+To create TimeLineFrames, use createKeyFrame(date: number, nodeCollection: NodeCollectionBase, nodeAppearance: NodeAppearance)
+- date - Date reference for the TimeLineFrame. Should provide the date value by Date.now() or number of milliseconds elapsed since January 1, 1970.
+- nodeCollection - Set of nodes to be rendered for the TimeFrame.
+- nodeAppearance - Styling of the nodes.
+
+### Play TimeLine
+We use play(durationInMilliSeconds: number, startDate: number, endDate: number)
+- durationInMilliSeconds - Number of milli-seconds after which next TimeLineFrame is rendered.
+- startDate - TimeLineFrame date to start the Playback of TimeLineFrames.
+- endDate - TimeLineFrame date to stop the Playback of TimeLineFrames.
+
+```js runnable
+// import { NodeCollectionBase, DefaultNodeAppearance, PropertyFilterNodeCollection } from '@cognite/reveal';
+// import { TimeLineTool } from '@cognite/reveal/tools';
+
+model.setDefaultNodeAppearance(DefaultNodeAppearance.Ghosted);
+
+const timeline = new TimeLineTool(viewer, model);
+
+const nodes = new AssetNodeCollection(sdk, model);
+timeline.createKeyFrame(Date.now(), nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Cyan });
+
+nodes.executeFilter({ assetId: 4222791488928479 });
+timeline.createKeyFrame(20000, nodes, { renderGhosted: false, outlineColor: NodeOutlineColor.Orange });
+
+const allPipes = new PropertyFilterNodeCollection(sdk, model);
+allPipes.executeFilter({'PDMS': {'Type': 'PIPE'}});
+timeline.createKeyFrame(40000, allPipes, { renderGhosted: false, color: [40, 200, 20] });
+
+const stuMechNodes = new PropertyFilterNodeCollection(sdk, model);
+stuMechNodes.executeFilter({'PDMS': {':capDisc': 'MECH', 'Purpose': 'STU' }});
+timeline.createKeyFrame(60000, stuMechNodes, { renderGhosted: false, color: [80, 100, 120] });
+
+timeline.play(5000, 1000, 80000);
+
+```
```

Comment:
Use real dates in the example to better describe the use case.

Reply:
Made changes